src/core/js/app.js

```diff
@@ -179,24 +182,32 @@ require([
 
     function onConfigModelDataLoaded() {
         Adapt.config.on({
-            'change:_defaultLanguage': onDefaultLanguageChange,
-            'change:_defaultDirection': onDefaultDirectionChange
+            'change:_activeLanguage': onLanguageChange,
+            'change:_defaultDirection': onDirectionChange
         });
     }
 
-    function onDefaultLanguageChange(model, language) {
+    function onLanguageChange(model, language) {
         Adapt.offlineStorage.set('lang', language);
-        Adapt.loadCourseData(language);
+        Adapt.loadCourseData(true);
     }
 
-    function onDefaultDirectionChange(model, direction) {
+    function onDirectionChange(model, direction) {
         $('html').toggleClass('dir-rtl', direction === 'rtl');
     }
 
+    function onConfigModelLoadCourseData() {
```

Comment:
could you just add this comment?
 /**
 * Before we actually go to load the course data, we first need to check to see if a language has been set.
 * If it has we can go ahead and start loading; if it hasn't, apply the defaultLanguage from config.json  
 */